yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1467,21 +1467,32 @@ def off_axis_projection_SPH(np.float64_t[:] px,
                             np.float64_t[:] quantity_to_smooth,
                             np.float64_t[:, :] projection_array,
                             normal_vector,
+                            north_vector,
                             weight_field=None):
     # Do nothing in event of a 0 normal vector
     if np.allclose(normal_vector, np.array([0., 0., 0.]), rtol=1e-09):
         return
 
+    # We want to do two rotations, one to first rotate our coordinates to have
+    # the normal vector be the z-axis (i.e., the viewer's perspective), and then
+    # another rotation to make the north-vector be the y-axis (i.e., north).
+    # Fortunately, total_rotation_matrix = rotation_matrix_1 x rotation_matrix_2
     cdef int num_particles = np.size(px)
-    cdef np.float64_t[:, :] rotation_matrix = get_rotation_matrix(normal_vector)
+    cdef np.float64_t[:] z_axis = np.array([0., 0., 1.], dtype='float_')
+    cdef np.float64_t[:] y_axis = np.array([0., 1., 0.], dtype='float_')
+    cdef np.float64_t[:, :] normal_rotation_matrix = get_rotation_matrix(normal_vector, z_axis)
+    cdef np.float64_t[:] transformed_north_vector = np.matmul(normal_rotation_matrix, north_vector)
+    cdef np.float64_t[:, :] north_rotation_matrix = get_rotation_matrix(transformed_north_vector, y_axis)
+    cdef np.float64_t[:, :] rotation_matrix = np.matmul(north_rotation_matrix, normal_rotation_matrix)
```

Comment:
Could we move these definitions to look more like this?
```python
cdef np.float64_t[:, :] normal_rotation_matrix
cdef np.float64_t[:] transformed_north_vector
cdef np.float64_t[:, :] north_rotation_matrix
cdef np.float64_t[:, :] rotation_matrix

normal_rotation_matrix = get_rotation_matrix(normal_vector, z_axis)
transformed_north_vector = np.matmul(normal_rotation_matrix, north_vector)
north_rotation_matrix = get_rotation_matrix(transformed_north_vector, y_axis)
rotation_matrix = np.matmul(north_rotation_matrix, normal_rotation_matrix)
```
I think it makes these lines a little easier on the eye

Reply:
I was just following the code style that you originally wrote (line 1476), but sure, I can change it.

Reply:
I see! None of this code is mine actually, I haven't really worked on this code. This is another summer student that Nathan had who implemented all of the cool projection stuff.